File: 2015/01/p2.cpp
// Day 1: Not Quite Lisp (Part 2) https://adventofcode.com/2015/day/1#part2
#include <bits/stdc++.h>

using namespace std;

int main(void) {
    string input = "";
    cin >> input;

    int floor = 0;

    for (int i = 0; i < input.length(); i++) {
        if (input[i] == '(') {
            floor++;
        } else {
            floor--;
        }

        if (floor == -1) {
            cout << i + 1 << endl;
            break;
        }
    }

    return 0;
}

File: 2015/08/p2.py
# Day 8: Matchsticks (Part 2) https://adventofcode.com/2015/day/8#part2
from re import sub

def real_len(string: str) -> int:

    string = string.replace("\\", "\\\\")
    string = string.replace("\"", "\\\"")
    string = f"\"{string}\""

    return len(string)


normal_string: int = 0
expanded_string: int = 0

while True:
    try:
        line: str = input().strip()
    except EOFError:
        break

    normal_string += len(line)
    expanded_string += real_len(line)

print(expanded_string - normal_string)
